### ISSUE TITLE
`C_ScenarioInfo.GetCriteriaInfo` doesn't provide quantityString anymore

### ISSUE THREAD

**Issue by LiangYuxuan** [Stanzilla/WoWUIBugs#592]
`C_ScenarioInfo.GetCriteriaInfo` is new function replacing [`C_Scenario.GetCriteriaInfo`](https://warcraft.wiki.gg/wiki/API_C_Scenario.GetCriteriaInfo) in TWW, which returns table now. However, it provides all old returns but `quantityString`.

The field `quantity` is actually percentage value with `isWeightedProgress` is true, which is the same behavior with `C_Scenario.GetCriteriaInfo`. The raw quantity value is exposed by `quantityString` and addons can parse from it. So the missing field making raw quantity value no longer possible available for weighted progress like _Enemy Force_ in M+.

```
Dump: value=C_ScenarioInfo.GetCriteriaInfo(4)
[1]={
  criteriaID=0,
  description="敌方部队",
  elapsed=0,
  duration=0,
  isWeightedProgress=true,
  completed=false,
  quantity=2,
  isFormatted=false,
  failed=false,
  assetID=0,
  flags=0,
  criteriaType=0,
  totalQuantity=260
}
```

It would be better if the function could accept new argument to provide raw quantity value directly.

**Comment by Numynum** [comment]
This blocks addons that provide more significant digits for the m+%, or addons that provide raw progress count information

Solutions include passing an argument as suggested by OP, but adding some `rawCount` field to the response would equally solve this.

A presumably less desirable alternative is to revert to previous behavior: return the raw count, and once more calculate the % in the UI itself

**Comment by Stanzilla** [comment]
This will be hotfixed in the next few days

**Comment by Numynum** [comment]
Any info what the change will be?
great news anyway 🥳

**Comment by Meorawr** [comment]
Fixed in 11.0.2.56162 ("quantityString" is part of the criteria info structure).

**Comment by Ex-wind** [comment]
> Fixed in 11.0.2.56162 ("quantityString" is part of the criteria info structure).

![API2](https://github.com/user-attachments/assets/dd7c7c05-bbc6-4021-af16-b26ab5deaadc)
quantityString= "9%"

Although I just need to write one more line of code to fix it

local quantity = tonumber(string.match(quantityString, "%d+"))
